### ISSUE TITLE
Make cell currents a model output

### ISSUE THREAD

**Issue by blakecaldwell** [jonescompneurolab/hnn-core#187]
HNN GUI saves cell currents for 'L2Pyr_soma' and  'L5Pyr_soma' to the file `i.txt`, but doesn't currently plot them. Adding the functionality to hnn-core to calculate these currents would allow us to explore ways to represent/plot this data in the future.

Function to record currents (called for L2Pyr and L5Pyr):
https://github.com/jonescompneurolab/hnn/blob/bd2abe4007160a5186856bc17dd87ae62055b383/cell.py#L180-L194

Needs to be aggregated for each type of cell: https://github.com/jonescompneurolab/hnn/blob/bd2abe4007160a5186856bc17dd87ae62055b383/network.py#L370-L387

The above function is called after a simulation completes:
https://github.com/jonescompneurolab/hnn/blob/d4c4cf94aa1db8517b99b0e344aa852c16125f50/run.py#L435

After this functionality exists to retrieve currents as a model output (same mechanism as somatic voltages?), the code to save the output to `i.txt` can be added back to HNN GUI. It would be a cleaner version of this code:
https://github.com/jonescompneurolab/hnn/blob/d4c4cf94aa1db8517b99b0e344aa852c16125f50/run.py#L140-L145

**Comment by ntolley** [comment]
#190 Is very similar with somatic voltages. It is not clear to me if this should be included in the same PR. The end goal is for cell level information to be accesible in a clean API as @jasmainak laid out: 
```py
>>> net = Network(...)
>>> net.cells[50].spikes
>>> net.plot_spikes()
>>> `net.cells[50].vsoma
>>> plt.plot(net.cells[50].vsoma)
```

**Comment by jasmainak** [comment]
I anticipate the diff in #190 to become large-ish. So, in the interest of an easier review, I suggest deferring this to another PR. You'll know what to do after #190 :-)

**Comment by blakecaldwell** [comment]
@ntolley can you update this your thoughts on moving forward with this? I know it is not required for hnn integration, but was curious if you had plans for adding this at a later time?

**Comment by ntolley** [comment]
The last commit to the somatic currents PR achieves this goal. It's implemented in largely the same way as the voltages such that `spikes.isoma` is a dictionary indexed by gids. 

There were several improvement ideas that came up in the somatic voltages PR. We were thinking of tacking them onto the somatic currents PR. If you were interested in incorporating currents into the GUI I can open a separate PR specifically for improvements to the `Spikes` class. 

Any thoughts @jasmainak? It might make sense to just have a coherent discussion on the new `Spikes` API in an independent PR.

**Comment by blakecaldwell** [comment]
I see, thanks for catching me up. I don't want to go crazy adding more things to the GUI. So I may check that the API is sufficient and make suggestions. That's fine to do in a separate PR.

We can let the use case for somatic currents in hnn-core drive what (if any) visualization features should be added to the GUI.

**Comment by jasmainak** [comment]
yeah I agree with @blakecaldwell that let's get MVP for the GUI. 

For the `Spikes` PR let's do simple and easy stuff first -- rename to `CellResponse` seems straightforward, add test to check agreement between `spiketimes` and voltages etc and maybe clean up some of the functions. Your to-to list looks about right. Smaller PR will be easier and faster to merge. So ideally keeping diff under 200-300 lines (red + green) helps.

**Comment by ntolley** [comment]
Sounds good to me. Once it's merged the most logical thing to tackle is storing `CellResponse` as a list. I'm going to wait until that's done before adding new visualization as that will require some pretty big changes to the existing member functions of `Spikes`. For example, should `mean_rates()` be moved to the `Network` class if `CellResponse` is stored as a list? 

In any case we can discuss the details once that PR has been started.
